### ISSUE TITLE
Linked heading has undesirable font in toctree

### ISSUE THREAD

**Issue by barbara-sfx** [sphinx-doc/sphinx#5391]
Subject: Linked heading has undesirable font in toctree

### Problem
I am experimenting with making some of our headings be links. They show up in the TOC sidebar in the font used for the :any: links in our docs. Is there a way to tell sphinx to use the standard font in the sidebar TOC?

## Actual Result

Here's what the TOC looks like:

![image](https://user-images.githubusercontent.com/17709550/45116993-3f529980-b109-11e8-86fc-f4ee347f4c2e.png)

#### Procedure to reproduce the problem
```
test heading with link:

:ref:`Amazon Web Services<monitor-aws>`
-------------------------------------------------------------------

```

#### Expected results

Heading should look like the one right above it in the TOC

### Environment info
- OS: Mac High Sierra
- Python version: 2.7.11
- Sphinx version: 1.7.5

**Comment by tk0miya** [comment]
Note: the behavior has started since d60ca8d49d. It seems there are no special meaning to represent references by literal.

**Comment by tk0miya** [comment]
Fixed by #6126.
Thank you for reporting!
